pkg/workloads/console/controller.go

```diff
@@ -74,115 +74,113 @@ func Add(ctx context.Context, logger kitlog.Logger, mgr manager.Manager, opts ..
 	logger   kitlog.Logger
 	recorder record.EventRecorder
 	client   client.Client
 }
 
-func (r *ConsoleReconciler) Reconcile(request reconcile.Request) (res reconcile.Result, err error) {
-	logger := kitlog.With(r.logger, "request", request)
+func (c *Controller) Reconcile(request k8rec.Request) (k8rec.Result, error) {
+	logger := kitlog.With(c.logger, "request", request)
 	logger.Log("event", EventStart)
 
-	defer func() {
-		if err != nil {
-			logger.Log("event", EventError, "error", err)
-		}
-	}()
-
-	name := request.NamespacedName
 	csl := &workloadsv1alpha1.Console{}
-	if err := r.client.Get(r.ctx, name, csl); err != nil {
+	if err := c.client.Get(c.ctx, request.NamespacedName, csl); err != nil {
 		if errors.IsNotFound(err) {
-			return res, logger.Log("event", EventNotFound, "resource", Console)
+			logger.Log("event", EventNotFound, "resource", Console)
 		}
-		return res, err
+		return k8rec.Result{}, err
 	}
 
-	logger = logging.WithRecorder(logger, r.recorder, csl)
+	// This is temporarily disabled as our logs don't pass k8s event validation
+	// logger = logging.WithRecorder(logger, c.recorder, csl)
 
+	reconciler := &ConsoleReconciler{
+		ctx:     c.ctx,
+		logger:  logger,
+		client:  c.client,
+		name:    request.NamespacedName,
+		console: csl,
+	}
+	result, err := reconciler.Reconcile()
+	if err != nil {
+		logger.Log("event", EventError, "error", err)
+	}
+	return result, err
+}
+
+type ConsoleReconciler struct {
+	ctx     context.Context
+	logger  kitlog.Logger
+	client  client.Client
+	name    types.NamespacedName
+	console *workloadsv1alpha1.Console
+}
+
+func (r *ConsoleReconciler) Reconcile() (res k8rec.Result, err error) {
 	// Fetch the console template
 	consoleTemplateName := types.NamespacedName{
-		Name:      csl.Spec.ConsoleTemplateRef.Name,
-		Namespace: name.Namespace,
+		Name:      r.console.Spec.ConsoleTemplateRef.Name,
+		Namespace: r.name.Namespace,
 	}
 	consoleTemplate := &workloadsv1alpha1.ConsoleTemplate{}
 	if err := r.client.Get(r.ctx, consoleTemplateName, consoleTemplate); err != nil {
 		if errors.IsNotFound(err) {
-			logger.Log("event", EventNotFound, "resource", ConsoleTemplate)
+			r.logger.Log("event", EventNotFound, "resource", ConsoleTemplate)
 		}
 		return res, err
 	}
 
 	// Try to find the job
-	jobExists := false
 	job := &batchv1.Job{}
-	err = r.client.Get(r.ctx, name, job)
+	if err := r.client.Get(r.ctx, r.name, job); err != nil && !errors.IsNotFound(err) {
+		return res, err
+	}
 
-	if err != nil {
-		if !errors.IsNotFound(err) {
+	jobExists := err != nil && !errors.IsNotFound(err)
+
+	// If the job hasn't expired, ensure it exists and is up to date
+	if !isJobExpired(r.console) {
+		job = buildJob(r.name, consoleTemplate.Spec.Template)
+		if err := r.createOrUpdate(job, Job, jobDiff); err != nil {
 			return res, err
 		}
-	} else {
-		// Clear the 'not found' error, as we don't want this to cause a retry of
-		// the reconciliation.
-		err = nil
 		jobExists = true
```

Comment:
I really wanted to avoid having to keep track of this variable throughout reconciliation, but I couldn't see a clean way to do it on this pass. I think it's something we should revisit if we can.